### PR TITLE
junit 4 assignment

### PR DESCRIPTION
**Pull request by jasleenwahi**

### CODE REVIEW

File: junit4assignment/src/main/java/org/example/StringUtils.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        char[] arrayOfStringToReverse = stringToReverse.toCharArray();
+        String reversedString= "";
+        for(Integer index = arrayOfStringToReverse.length-1; index>=0; index--)
+            reversedString = reversedString+arrayOfStringToReverse[index];
+
+        return reversedString;
+    }
+
+    // function used to check if string is empty or not.
+    public Boolean isEmpty1(String stringToCheck)
```
Comment (anup0060) [inline junit4assignment/src/main/java/org/example/StringUtils.java]: poor method naming

File: junit4assignment/src/test/java/org/example/StringUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    StringUtils stringUtils;
+    @Before
+    public void setup()
+    {
+        stringUtils = new StringUtils();
+    }
+
+    // testing reverse method with non empty string.
+    @Test
+    public void testReverseShouldReturnReverse_test1()
```
Comment (anup0060) [inline junit4assignment/src/test/java/org/example/StringUtilsTest.java]: _test1 _test2 , why all methods with these suffixes 

testReverse_givenExpectedInput -- like this, we need to have the name that reflects scenario as suffix.

**Review by anup0060 — CHANGES_REQUESTED** [review]

**Review by jasleenwahi — COMMENTED** [review]
changes made requested by you, thank you for the review it helped me learn a lot.

File: junit4assignment/src/test/java/org/example/StringUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    StringUtils stringUtils;
+    @Before
+    public void setup()
+    {
+        stringUtils = new StringUtils();
+    }
+
+    // testing reverse method with non empty string.
+    @Test
+    public void testReverse_The_String_With_NonEmpty_String()
```
Comment (anup0060) [inline junit4assignment/src/test/java/org/example/StringUtilsTest.java]: see review comment of junit 5 , apply here - naming convention

**Review by anup0060 — CHANGES_REQUESTED** [review]

**Review by jasleenwahi — COMMENTED** [review]
made all the required changes, thankyou for the review sir

**Review by jasleenwahi — COMMENTED** [review]
Changed the convention, thank you for making me understand better naming convention

**Review by anup0060 — APPROVED** [review]

**Pull request merged**